src/infra/dex/paraswap/mod.rs

```diff
@@ -50,73 +50,41 @@ impl ParaSwap {
         }
     }
 
+    /// Make a request to the `/swap` endpoint.
     pub async fn swap(
         &self,
         order: &dex::Order,
-        slippage: &dex::Slippage,
+        slippage: dex::Slippage,
         tokens: &auction::Tokens,
     ) -> Result<dex::Swap, Error> {
-        let price = self.price(order, tokens).await?;
-        let transaction = self.transaction(&price, order, tokens, slippage).await?;
+        let query = dto::SwapQuery::new(&self.config, order, tokens, slippage)?;
+        let swap = util::http::roundtrip!(
+            <dto::Swap, dto::Error>;
+            self.client.request(reqwest::Method::GET, util::url::join(&self.config.endpoint, "swap"))
+                .header("X-API-KEY", &self.config.api_key)
```

Comment:
Setting the header is no longer needed with https://github.com/gnosis/solvers/pull/32, right?